tests/test-recipes/metadata/noarch_foo/meta.yaml

```diff
@@ -0,0 +1,6 @@
+package:
+  name: noarch_foo_test_package
+  version: "1.0"
+
+build:
+  noarch: foo
```

Comment:
I think it would be good to add the test for this, so that you verify that the metadata accurately reflects your setting here.  I think you might need to break this test out (add an _ to the folder name) and manually run the test, like you do for the noarch_none_value.

The test probably looks something like:

```
def sometest(test_config):
    fn = api.get_output_file_path(recipe_path, config=test_config)
    api.build(recipe_path, config=test_config)
    metadata = json.loads(package_has_file(fn, 'index.json'))
    assert 'noarch' in metadata
    assert metadata['noarch'] == foo
```

I think index.json is the right file, but it might be info.json.